src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -1379,11 +1379,22 @@ void Convolution::prepareParams() {
             auto reordItpd = reorderConvDesc->createPrimitiveDescriptorIterator(engine, key.attr);
             if (static_cast<bool>(reordItpd)) {
                 auto prim_desc = convolution_forward::primitive_desc(reordItpd.get());
+                MemoryPtr weightMemPtr;
+                dnnl::memory::desc wghDesc = key.inp1->getDnnlDesc();
+                auto wEdgePtr = this->getParentEdgesAtPort(1)[0];
+                if (wEdgePtr->getParent()->isConstant()) {
+                    // reorder constant weights once only and keep it in cache
+                    wghDesc = prim_desc.weights_desc();
+                    weightMemPtr = this->reorderWeightForSharing(*wEdgePtr->getMemoryPtr(),
+                                                                 0,
+                                                                 DnnlExtensionUtils::makeDescriptor(wghDesc));
+                }
```

Comment:
As It was said above, we need some another approach on how to handle the weight caching.
What do you think about the fallowing?
Let's pass to the ConvolutionExecutor object a reference to the weights cache. All we need to do, is, in case of constant, when the existing and required by the implementation tensor descriptors are not match, instead of creating a reorder, we store some info that we need to solve the mismatch leter. Then, at the exec call to the executor, we should call thomething like 'reorderWeightForSharing' to get from cache o create a new reordered weights blob. The problem is that we need to do it quick, so that no to apply the overhead on the exec call. To this end we have to define some lightweight key for the reordered weights tensors. Something simple, like in 'Edge::externalAllocate' So the external interfaces will remains the same, instead of some changed or new constructors. All the work will be performed inside the dnnl executor implementation. Or, may be inside the convolution implementation for the beginnig.

Reply:
I did some change to the design, please review again, Thanks!